### PR TITLE
Add date custom element & processor

### PR DESCRIPTION
**Pull request by emma-sg**
~~Haven't really tested this — @lun-4 would u mind?~~

tested locally & it works !

### CODE REVIEW

File: src/resources/at-date.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    time.setAttribute("datetime", dateText);
+    const slot = document.createElement("slot");
+
+    time.appendChild(slot);
+    shadow.appendChild(time);
+
+    const stylesheet = new CSSStyleSheet();
+
+    stylesheet.replace(`
+      :host {
```
Comment (lun-4) [inline src/resources/at-date.js:25]: why can't we put this on `styles.css`?

Comment (emma-sg) [reply src/resources/at-date.js:25]: cause it's attached to the custom element - the shadow dom means you can't (easily) write css in the root element that penetrates into custom elements. also cause i'm setting `content` prop on `:host:after` based on component props

Comment (lun-4) [reply src/resources/at-date.js:25]: fuckd up...

**Review by lun-4 — APPROVED** [review]
:shipit:

**Pull request merged**